src/main/java/bio/terra/service/resourcemanagement/google/GoogleResourceService.java

```diff
@@ -68,44 +77,148 @@ public GoogleResourceService(
+        GoogleBucketResource googleBucketResource = resourceDao.createAndLockBucket(bucketRequest, flightId);
+
+        boolean metadataCreationFailed = (googleBucketResource == null);
+        logger.info("metadataCreationFailed = " + metadataCreationFailed);
+        if (metadataCreationFailed) {
+            // insert failed. lookup the existing bucket_resource row
+            googleBucketResource = resourceDao.getBucket(bucketRequest);
+        }
+
+        // this fault is used by the ResourceLockTest
+        if (configService.testInsertFault(ConfigEnum.BUCKET_LOCK_CONFLICT_STOP_FAULT)) {
+            try {
+                logger.info("BUCKET_LOCK_CONFLICT_STOP_FAULT");
+                while (!configService.testInsertFault(ConfigEnum.BUCKET_LOCK_CONFLICT_CONTINUE_FAULT)) {
+                    logger.info("Sleeping for CONTINUE FAULT");
+                    TimeUnit.SECONDS.sleep(5);
+                }
+                logger.info("BUCKET_LOCK_CONFLICT_CONTINUE_FAULT");
+            } catch (InterruptedException ex) {
+                Thread.currentThread().interrupt();
+                throw new LoadLockFailureException("Unexpected interrupt during bucket lock fault");
             }
-        } catch (GoogleResourceNotFoundException e) {
-            logger.info("no bucket resource metadata found for project: {}", projectResource.getGoogleProjectId());
         }
 
-        // the bucket might already exist even though we don't have metadata for it
-        Bucket bucket = getBucket(bucketRequest.getBucketName()).orElseGet(() -> newBucket(bucketRequest));
+        String bucketName = bucketRequest.getBucketName();
+        String lockingFlightId = googleBucketResource.getFlightId();
+        logger.info("lockingFlightId = " + lockingFlightId);
+        Bucket bucket;
+        if (lockingFlightId == null) {
+            // GET case. the row in the bucket_resource table is unlocked. the bucket has already been created
+
+            // throw an exception if the bucket does not already exist
+            bucket = getBucket(bucketName)
+                .orElseThrow(() -> new GoogleResourceNotFoundException("Bucket not found: " + bucketName));
+        } else if (lockingFlightId.equals(flightId)) {
+            // CREATE case. the row in the bucket_resource table is locked by this flight. we need to create it here
+
+            // check if the bucket already exists
+            Optional<Bucket> existingBucket = getBucket(bucketName);
+            if (!existingBucket.isPresent()) {
+                // bucket DOES NOT EXIST. create it and unlock
+                bucket = newBucket(bucketRequest);
+                resourceDao.unlockBucket(bucketName, flightId);
+            } else {
+                // bucket EXISTS. if this is a recovery flight or testing environment, use it. otherwise throw exception
+                boolean springProfileIsForTesting = Arrays.stream(springEnvironment.getActiveProfiles()).anyMatch(
```

Comment:
I think the recommended way to do this would be to have a prop in `application-<profile>.properties` that we check to determine whether we are in strict mode or not.

Reply:
Added a new application property datarepo.gcs.allowReuseExistingBuckets. Default is false, set it to true in the integration, integrationtest, and connectedtest properties files. Thanks for the suggestion.